src/main/java/ro/msg/learning/shop/dto/ErrorMissingStockDTO.java

```diff
@@ -0,0 +1,18 @@
+package ro.msg.learning.shop.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import ro.msg.learning.shop.exceptions.NoSuitableLocationsFound;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class ErrorMissingStockDTO {
+    private String type;
```

Comment:
where is type used?

Reply:
"type" is used for implementing the static factory method pattern => the appropiate "of()-method" in class "ErrorMissingStockDTO.class"